### ISSUE TITLE
`maturin init` then `develop` gives `error: failed to select a version for the requirement`

### ISSUE THREAD

**Issue by reece** [PyO3/maturin#759]
### Bug Description

When trying to create a new cargo project using `maturin init`, then trying to build with `maturin develop`, I get `error: failed to select a version for the requirement`. `cargo build` fails.

I am brand new to RUST, pyo3, and maturin, so I might be doing something incorrectly.

```
(3.9) snafu$ python3 -V
Python 3.9.7

(3.9) snafu$ maturin -V
maturin 0.12.5

(3.9) snafu$ rustc -V
rustc 1.53.0

(3.9) snafu$ lsb_release -a
No LSB modules are available.
Distributor ID:	Ubuntu
Description:	Ubuntu 21.10
Release:	21.10
Codename:	impish

```

### Your Python version (`python -V`)

3.9.7

### Your pip version (`pip -V`)

21.3.1

### What bindings you're using

pyo3

### Does `cargo build` work?

- [ ] Yes, it works

### If on windows, have you checked that you aren't accidentally using unix path (those with the forward slash `/`)?

- [ ] Yes

### Steps to Reproduce

```
bash-5.1$ cd /tmp
python3.9 -m venv venv/3.9
source venv/3.9/bin/activate
pip install -U setuptools pip
pip install -U maturin
maturin init -b pyo3 maturin-test
cd maturin-test
maturin develop
```

gives:

```
Requirement already satisfied: setuptools in ./venv/3.9/lib/python3.9/site-packages (44.1.1)
Collecting setuptools
  Using cached setuptools-60.2.0-py3-none-any.whl (953 kB)
Requirement already satisfied: pip in ./venv/3.9/lib/python3.9/site-packages (20.3.4)
Collecting pip
  Using cached pip-21.3.1-py3-none-any.whl (1.7 MB)
Installing collected packages: setuptools, pip
  Attempting uninstall: setuptools
    Found existing installation: setuptools 44.1.1
    Uninstalling setuptools-44.1.1:
      Successfully uninstalled setuptools-44.1.1
  Attempting uninstall: pip
    Found existing installation: pip 20.3.4
    Uninstalling pip-20.3.4:
      Successfully uninstalled pip-20.3.4
Successfully installed pip-21.3.1 setuptools-60.2.0
Collecting maturin
  Using cached maturin-0.12.5-py3-none-manylinux_2_5_x86_64.manylinux1_x86_64.whl (8.9 MB)
Collecting toml~=0.10.2
  Using cached toml-0.10.2-py2.py3-none-any.whl (16 kB)
Installing collected packages: toml, maturin
Successfully installed maturin-0.12.5 toml-0.10.2
  ✨ Done! Initialized project maturin-test
💥 maturin failed
  Caused by: Cargo metadata failed. Does your crate compile with `cargo build`?
  Caused by: `cargo metadata` exited with an error:     Updating crates.io index
error: failed to select a version for the requirement `pyo3 = "^0.15.1"`
candidate versions found which didn't match: 0.14.5, 0.14.4, 0.14.3, ...
location searched: crates.io index
required by package `maturin-test v0.1.0 (/tmp/maturin-test)`
```

```
(3.9) bash-5.1$ cargo build
    Updating crates.io index
error: failed to select a version for the requirement `pyo3 = "^0.15.1"`
candidate versions found which didn't match: 0.14.5, 0.14.4, 0.14.3, ...
location searched: crates.io index
required by package `maturin-test v0.1.0 (/tmp/maturin-test)`
```

**Comment by messense** [comment]
pyo3 0.15.1 does exist on crates.io: https://crates.io/crates/pyo3/0.15.1

Check your network connection maybe?

**Comment by messense** [comment]
How did you install Rust? Looks like it's kinda old. Try install Rust using rustup instead: https://rustup.rs/

**Comment by reece** [comment]
I installed with apt install. rust 1.53.0 is the only version available in Ubuntu 21.10.  Network is fine.

I noticed `~/.cargo`. I moved it out of the way and then `maturin develop` worked fine. It appears that something about my previous (and old) `~/.cargo` that caused this issue. The old directory contains .crates2.json, .crates.toml, and bin/.

In any case, now solved. Thanks for following up.



```
[...]
💥 maturin failed
  Caused by: Cargo metadata failed. Does your crate compile with `cargo build`?
  Caused by: `cargo metadata` exited with an error:     Updating crates.io index
error: failed to select a version for the requirement `pyo3 = "^0.15.1"`
candidate versions found which didn't match: 0.14.5, 0.14.4, 0.14.3, ...
location searched: crates.io index
required by package `maturin-test v0.1.0 (/tmp/maturin-test)`

(3.9) bash-5.1$ mv ~/.cargo/ ~/tmp/
(3.9) bash-5.1$ maturin develop
🔗 Found pyo3 bindings
🐍 Found CPython 3.9 at /tmp/venv/3.9/bin/python
   Compiling proc-macro2 v1.0.36
   Compiling unicode-xid v0.2.2
[...]
   Compiling maturin-test v0.1.0 (/tmp/maturin-test)
    Finished dev [unoptimized + debuginfo] target(s) in 10.06s
📦 Built wheel for CPython 3.9 to /tmp/maturin-test/target/wheels/maturin_test-0.1.0-cp39-cp39-linux_x86_64.whl
🛠  Installed maturin-test-0.1.0
```
